### ISSUE TITLE
API: no endpoint to fetch `nonce` when pending transctions

### ISSUE THREAD

**Issue by witoldsz** [semuxproject/semux#900]
As of now, we have an `/account?address=…` endpoint. It returns `nonce`, but it ignores pending transactions:

```
{
    "address": "0x7acb178cfb49b84ae51f79987a8eda1c43b712b0",
    "available": "20975000000",
    "locked": "1000000000",
    "nonce": "5",
    "transactionCount": 23,
    "pendingTransactionCount": 1
  }
```

I was sure I could just add: `nonce+pendingTransactionCount`, but `pendingTransactionCount` does not mean the current `nonce` will be incremented, because pending transactions are also incoming ones and these do not alter `nonce`.

Another idea was to fetch `/accounts/pending-transactions`, but as above: you can have many transactions there and one has to fetch them all and search for the last one having `from` equal the address I need to get the `nonce` for.

Long story short: if you want to create a transaction, you have to:
- get `/account?address=X`
- if the above has `pendingTransactions: 0` then you take `nonce`
- if not, you have to fetch all `/account/pending-transactions` and search for the last one having `from=X`,
- finally you take the `nonce` from that

Seems quite a lot of hassle.

That would be great if `/account` has `pendingNonce`.

**Comment by orogvany** [comment]
Would prefer not a new endpoint, as that would still be 2 calls to do it (I assume you need other information from account call, so it's just swapping pending for nonce call)

**Comment by witoldsz** [comment]
> so it's just swapping pending for nonce call

It's not just swapping endpoint X to Y. As I wrote above, 
- you have to fetch account to get the number of pending transactions 
- then fetch all pending transactions of given account 
- then you have to filter the list and 
- then you have to take last element. 

If there are lots of pending transactions then it's obviously a waste.

> Would prefer not a new endpoint, as that would still be 2 calls to do it

If `/account/nonce` was returning nonce and pendingNonce, then one call would do. No need for `/account` at that point since all the data to display the form must be fetched earlier anyway.

**Comment by orogvany** [comment]
Oh sorry i wasnt clear.  I meant that youd still need to make 2 calls with nonce call.  Doing a for loop over pending is trivial.  Currently transaction volume is so low, youre not going to get any improvement here in practice.

Was just noting that 2 calls is still 2 calls.  Nothing more
